### PR TITLE
Fix exported images

### LINKED ISSUE #28: Images are not exported properly
The images in the export of the module do not export the images properly. It can happen that there are no valid images in the database and when iterating over the image-objects the foreach breaks and a misleading URL is exported.

Maybe we could use the standard functionality of Magento to get the images

### PR DESCRIPTION
**Pull request by fin-markus**
This will fix #28

### CODE REVIEW

File: app/code/community/Findologic/Search/Model/ProductImage.php [hunk 1]
```diff
@@ -37,10 +37,11 @@ public function getBaseImageUrlByProductId($productId)
     {
         $product =  Mage::getModel('catalog/product')->load($productId);
         $productMediaConfig = Mage::getModel('catalog/product_media_config');
+        $productImage = $product->getImage();
         $baseImageUrl = '';
 
-        if ($product->getImage() != 'no_selection') {
-            $baseImageUrl = $productMediaConfig->getMediaUrl($product->getImage());
+        if ($productImage && $productImage != 'no_selection') {
```
Comment (ihanli) [inline app/code/community/Findologic/Search/Model/ProductImage.php]: `!=` -> `!==`

File: app/code/community/Findologic/Search/Helper/XmlBasic.php [hunk 2]
```diff
@@ -208,20 +208,25 @@ public function renderImages(SimpleXMLElement $allImages)
...
             foreach ($productImages as $productImage) {
                 if (self::$mediaHelper->getMediaUrl($productImage['value']) != $baseImageUrl) {
                     $this->appendCData($images->addChild('image'), self::$mediaHelper->getMediaUrl($productImage['value']));
+                    $hasValidImage = true;
                 }
             }
-        } else {
+        }
+
+        if(!$hasValidImage){
```
[the remark this answers is not in the document]
Comment (fin-markus) [reply app/code/community/Findologic/Search/Helper/XmlBasic.php]: Done

File: app/code/community/Findologic/Search/Model/ProductImage.php (hunk 1 above)
Comment (fin-markus) [reply app/code/community/Findologic/Search/Model/ProductImage.php]: Done

**Pull request merged**